server/src/main/java/org/opensearch/search/fetch/subphase/ScriptFieldsPhase.java

```diff
@@ -54,7 +54,7 @@ public final class ScriptFieldsPhase implements FetchSubPhase {
 
     @Override
     public FetchSubPhaseProcessor getProcessor(FetchContext context) {
-        if (context.scriptFields() == null) {
+        if (!context.hasScriptFields()) {
```

Comment:
Super nit, `if (context.hasScriptFields() == false) {` - this is just the convention we are trying to stick to across codebase

Reply:
Done!